src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/ShareholderService.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.companieshouse.confirmationstatementapi.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.confirmationstatementapi.client.OracleQueryClient;
+
+@Service
+public class ShareholderService {
+
+    @Autowired
+    private OracleQueryClient oracleQueryClient;
+
+    public int getShareholderCount(String companyNumber) {
```

Comment:
the Integer returned from getShareholderCount is being unboxed here into an int.
I would suggest returning an Integer from here and add a null check to count in the company shareholder validation